### PR TITLE
#1712 Bump to Polly 8.0

### LINKED ISSUE #1712: Bump to Polly 8.0
## Bump to Polly 8.0
and make some code improvements in Ocelot.polly package

### PR DESCRIPTION
**Pull request by RaynaldM**
## Closes #1712 
- #1712 

Bump to Polly 8.0 and some code improvements

### CODE REVIEW

File: src/Ocelot.Provider.Polly/PollyQoSProvider.cs [hunk 1]
```diff
@@ -39,25 +29,30 @@ public PollyQoSProvider(DownstreamRoute route, IOcelotLoggerFactory loggerFactor
...
             {
-                _circuitBreakerPolicy = null;
+                circuitBreakerPolicy = null;
             }
 
-            CircuitBreaker = new CircuitBreaker(_circuitBreakerPolicy, _timeoutPolicy);
+            CircuitBreaker = new CircuitBreaker(circuitBreakerPolicy, timeoutPolicy);
+        }
+
+        [Obsolete("do not use: it does nothing")]
```
Comment (ggnaegi) [inline src/Ocelot.Provider.Polly/PollyQoSProvider.cs:53]: Maybe you could mention use the other constructor instead, a bit more precise.

**Review by ggnaegi — APPROVED** [review]

**Comment by raman-m** [comment]
@RaynaldM Is it urgent or not?
Are you experiencing some problems with current /old version of Polly package in the project of your team?

I'm afraid it will be wrong to add this change to current release because of the package upgrading.
Let me know the urgency plz...

**Comment by RaynaldM** [comment]
> @RaynaldM Is it urgent or not? Are you experiencing some problems with current /old version of Polly package in the project of your team?
> 
> I'm afraid it will be wrong to add this change to current release because of the package upgrading. Let me know the urgency plz...

@raman-m no urgency currently

**Comment by raman-m** [comment]
> @RaynaldM commented [on Oct 2](#issuecomment-1743092457)

👌 I'm focusing on the release now...

**Comment by raman-m** [comment]
Don't worry about failed build [2086](https://circleci.com/gh/ThreeMammals/Ocelot/2086), this is  Scenario: [Open circuit should not effect different route](https://app.circleci.com/pipelines/github/ThreeMammals/Ocelot/1116/workflows/2225f603-791a-47db-ac2b-0960c00d7731/jobs/2086?invite=true#step-102-252948_56)
Next build should have green status. This is well-known bug-issue #1706 :
- #1706 

@ggnaegi  Hi! As you see, this test is constantly failing, rare, but it fails.
It seems we need to prioritize #1706 as **High** 😉 Do you have some time this week to spare on the bug?

**Comment by raman-m** [comment]
⚠️  Ray,
You are the member of the team. You can create feature branches right in this ThreeMammals/Ocelot repository 😉

File: src/Ocelot.Provider.Polly/PollyQoSProvider.cs [hunk 2]
```diff
@@ -39,25 +29,30 @@ public PollyQoSProvider(DownstreamRoute route, IOcelotLoggerFactory loggerFactor
...
+                circuitBreakerPolicy = null;
             }
 
-            CircuitBreaker = new CircuitBreaker(_circuitBreakerPolicy, _timeoutPolicy);
+            CircuitBreaker = new CircuitBreaker(circuitBreakerPolicy, timeoutPolicy);
+        }
+
+        [Obsolete("Use the constructor PollyQoSProvider(DownstreamRoute route, IOcelotLoggerFactory loggerFactory)")]
+        public PollyQoSProvider(AsyncCircuitBreakerPolicy circuitBreakerPolicy, AsyncTimeoutPolicy timeoutPolicy, IOcelotLogger logger)
+        {
```
Comment (raman-m) [inline src/Ocelot.Provider.Polly/PollyQoSProvider.cs:55]:
```suggestion
        private const string ObsoleteConstructorMessage = $"Use the constructor {nameof(PollyQoSProvider)}({nameof(DownstreamRoute)} route, {nameof(IOcelotLoggerFactory)} loggerFactory)!";

        [Obsolete(ObsoleteConstructorMessage)]
        public PollyQoSProvider(AsyncCircuitBreakerPolicy circuitBreakerPolicy, AsyncTimeoutPolicy timeoutPolicy, IOcelotLogger logger)
        {
            throw new NotSupportedException(ObsoleteConstructorMessage);
```
And it is strange that obsolete constructor has 0 (zero) references!...
Were there zero references?
If Not, all referenced scopes should be fixed, and addition to DI. But probably the constructor is not referenced at all.

**Review by raman-m — APPROVED** [review]

**Comment by raman-m** [comment]
Ray, I have refactored two classes. Are you OK with [these changes](#commits-pushed-97b9373)?

**Review by raman-m — APPROVED** [review]

**Review by ggnaegi — APPROVED** [review]

**Comment by raman-m** [comment]
@RaynaldM Are we going to merge? 😃

**Comment by raman-m** [comment]
@RaynaldM I will come back in 2 hours... Answer, please, are you OK to merge or not?

**Comment by RaynaldM** [comment]
> Ray, I have refactored two classes. Are you OK with [these changes](#commits-pushed-97b9373)?

Yes, I'm

**Comment by raman-m** [comment]
> @RaynaldM 
> Yes, I'm

How would you like to reuse this feature in your projects? Is it urgent to release?
Or are you building a local Ocelot solution from **develop** branch of your forked repo?

**Comment by RaynaldM** [comment]
We've released a 20.0.1-alpha on our artifacts repo, so we're in no hurry, it's still in the test phase.
We're sticking as closely as possible to the original Ocelot, and will continue to do so for as long as possible. So, we sometimes make temporary "alpha (or beta)" packages (e.g. migration to .Net 7) while waiting for integration into the official package.

**Comment by raman-m** [comment]
>@RaynaldM commented [on Oct 16 at 12:23 PM](#issuecomment-1764360885)
> We've released a 20.0.1-alpha on our artifacts repo, so we're in no hurry, it's still in the test phase...
> ...waiting for integration into the official package.

👌 We will have 20.0.1 official release these days because of [ReadTheDocs build](https://readthedocs.org/projects/ocelot/builds/) problem (the fix is merged into [release/20.0](../commits/release/20.0) branch already). If you want, I can include your feature from develop? Otherwise, you & your team will have to wait until the first days of November for the October's release.

**Comment by RaynaldM** [comment]
as you want @raman-m, for us, it's ok for begining of November

**Pull request merged**